caput/tests/test_mpiarray.py

```diff
@@ -394,7 +397,69 @@ def test_global_setslice(self):
 
         assert (darr == local_array).all()
 
+    def test_outer_ufunc(self):
+        rank = mpiutil.rank
+        size = mpiutil.size
+
+        dist_arr = mpiarray.MPIArray((size, 4), axis=0)
+        dist_arr[:] = rank
+
+        dist_arr_scalar = mpiarray.MPIArray((size, 4), axis=0)
+        dist_arr_scalar[:] = rank * 2
+
+        dist_arr_add = dist_arr + dist_arr
+
+        # Check that you can add two numpy arrays,
+        # if they are distributed along the same axes
+        # Check that you can multiple a numpy array against a scalar
+        assert (dist_arr_add == dist_arr_scalar).all()
```

Comment:
This test doesn't check the scalar case afaict. The `rank * 2` above is just a multiplication of two python scalars which are then assigned into the MPIArray.

I would also recommend separating out these two tests into separate asserts, i.e.
```suggestion
        dist_arr_add = dist_arr + dist_arr
        disr_arr_mul = 2 * dist_arr

        # Check that you can add two MPIArrays
        assert (dist_arr_add == 2 * rank).all()
        
        # Check that you can multiple an MPIArray against a scalar
        assert (dist_arr_sum == 2 * rank).all()
```

It's probably worth checking it works with differing arguments too (i.e. add different MPIArrays).

Also I think you were working on this, but we should test the exceptions in cases you expect to fail (i.e. different distributed axes, different shapes...),